### PR TITLE
implemented create location - FM-97

### PR DESCRIPTION
**Pull request by sashrika**

### CODE REVIEW

**Comment by sashrika** [comment]
there was a small bug in the util class , which was not setting the description correctly. I fix is also included.

File: omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java [hunk 1]
```diff
@@ -130,5 +134,22 @@ public MethodOutcome updateLocationById(@ResourceParam Location theLocation,
...
 	}
+	
+	/**
+	 * Create Location
+	 *
+	 * @param location fhir Location object
+	 * @return Method outcome object which contains the identity of the created resource.
+	 */
+	@Create
+	public MethodOutcome createFHIRPerson(@ResourceParam Location location) {
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java]: Method signature?

Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java]: I mean this method signature should changed it's in location resource

File: api/src/main/java/org/openmrs/module/fhir/api/impl/LocationServiceImpl.java [hunk 2]
```diff
@@ -135,5 +135,24 @@ public void updateLocationById(String id, Location location) {
...
+	/**
+	 * @see org.openmrs.module.fhir.api.LocationService#createLocation(ca.uhn.fhir.model.dstu2.resource
+	 * .Location)
+	 */
+	public Location createLocation(Location location) {
+		org.openmrs.Location omrsLocation=null;
+		List<String> errors = new ArrayList<String>();
+		omrsLocation = FHIRLocationUtil.generateOpenMRSLocation(location, errors);
+		if (!errors.isEmpty()) {
+			StringBuilder errorMessage = new StringBuilder("The request cannot be processed due to following issues \n");
```
Comment (surangak) [inline api/src/main/java/org/openmrs/module/fhir/api/impl/LocationServiceImpl.java]: "The request cannot be processed due to the following issues"

File: api/src/main/java/org/openmrs/module/fhir/api/impl/LocationServiceImpl.java [hunk 3]
```diff
@@ -135,5 +135,24 @@ public void updateLocationById(String id, Location location) {
...
+		List<String> errors = new ArrayList<String>();
+		omrsLocation = FHIRLocationUtil.generateOpenMRSLocation(location, errors);
+		if (!errors.isEmpty()) {
+			StringBuilder errorMessage = new StringBuilder("The request cannot be processed due to the following issues \n");
+			for (int i = 0; i < errors.size(); i++) {
+				errorMessage.append((i + 1) + " : " + errors.get(i) + "\n");
+			}
+			throw new UnprocessableEntityException(errorMessage.toString());
+		}
+		omrsLocation=Context.getLocationService().saveLocation(omrsLocation);
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/impl/LocationServiceImpl.java]: please make sure you are using proper code formatter. We are always leaves blank space before and after equal sign. Please address this in separate pull request.

File: api/src/test/java/org/openmrs/module/fhir/api/LocationServiceTest.java [hunk 4]
```diff
@@ -143,4 +143,18 @@ public void updateLocationById_shouldAddResourceIfNotExistsElseUpdateExistingRes
...
+	@Test
+	public void createOpenMRSLocation() throws Exception {
+		String locationUuid = "6f42abbc-caac-40ae-a94e-9277ea15c125";
+		org.openmrs.Location omrsLocation=Context.getLocationService().getLocationByUuid(locationUuid);
+		omrsLocation.setUuid("");// remove the uuid value from the Location. This will let this resource to be persist on the db with
+		// random uuid
+		Location fhirLocation = FHIRLocationUtil.generateLocation(omrsLocation);	
+		fhirLocation.setName("New Location Test Name");
+		fhirLocation = Context.getService(LocationService.class).createLocation(fhirLocation);
+		assertNotNull(fhirLocation);
```
Comment (harsha89) [inline api/src/test/java/org/openmrs/module/fhir/api/LocationServiceTest.java]: In a test method, asserting not null only not sufficient. Make sure you assert some attributes as well.

**Pull request merged**